browser/scripts/ui-core.js

```diff
@@ -67,7 +72,240 @@ VizorUI.prototype._init = function(e2) {	// called by .init() in ui.js
 	e2.core.on('progress', this.updateProgressBar.bind(this));
 };
 
+VizorUI.prototype.getZoom = function() {
+	return 1;
+}
+
+/**
+ * get the state of a UI (tabbed) panel. if no $domElement provided, then _found = false
+ * @param $domElement
+ * @returns {{_found: boolean, visible: boolean, collapsed: boolean, selectedTab: null, x: boolean, y: boolean}}
+ */
+VizorUI.prototype.getDomPanelState = function($domElement) {	/* @var $domElement jQuery */
+	var state = {
+		_found: false,
+		visible : true,
+		collapsed : false,
+		selectedTab : null,
+		x: false,
+		y: false,
+		w: -1,
+		h: -1
+	};
+	if ((typeof $domElement === 'object') && ($domElement.length > 0)) {
+		state._found = true;
+		state.visible = $domElement.is(':visible') && $domElement.hasClass('uiopen');
+		state.collapsed = $domElement.hasClass('collapsed');
+		var pos = $domElement.position();
+		if (pos) {
+			state.x = pos.left;
+			state.y = pos.top;
+		}
+		state.w = $domElement.width();
+		state.h = $domElement.height();
+		// get the active tab
+		var $activeLi = $domElement.find('li.active');
+		if ($activeLi.length>0) {
+			var tabName, tabLink;
+			tabName = $activeLi.data('tabname');			// data-tabname='presets' preferred
+			if (tabName) {
+				state.selectedTab = tabName;
+			} else {
+				tabLink = $activeLi.find('a');
+				state.selectedTab = tabLink.attr('href');	// #something
+			}
+		}
+		// else this isn't tabbed so selectedTab does not apply
+	}
+	return state;
+}
+
+VizorUI.prototype.getState = function() {
+	if (typeof clone !== 'function') return null;	// why are we here?
```

Comment:
since clone() is a dependency, this is just in case this is ever called without util.js in place.